// rollup-plugin-sass-migrator/sass-migrator.ts
import {spawnSync} from 'child_process'
import * as fs from 'fs'

export function sassMigrator(opts = {
  indexPath: undefined,
  dryRun: false,
  debug: false,
}) {
  if (!opts.indexPath) {
    console.error("The sass-migrator plugin requires a path to a SASS file.")
    return undefined
  }

  if (!fs.existsSync('node_modules/sass-migrator/sass-migrator.js')) {
    console.error("The sass-migrator plugin requires sass-migrator to be installed.\nTry running yarn add -D sass-migrator")
    return undefined
  }
  return {
    name: 'migrate-sass',
    buildStart: () => {
      if (opts.debug) {
        console.log(`migrate-sass [${opts.indexPath}]`)
      }
      const args = [
        'node_modules/sass-migrator/sass-migrator.js',
        'division',
        '--migrate-deps',
        opts.indexPath,
      ]
      if (opts.dryRun) {
        args.splice(-1, 0, "--dry-run")
      }
      const ret = spawnSync("node", args)
      if (ret.status) {
        console.warn(`There was an error migrating sass:\n${ret.stderr.toString().trim()}`)
      }
      if (opts.debug) {
        console.log(ret.stdout.toString().trim())
      }
    }
  }
}

export function sassMigratorQuasar(opts = {
  dryRun: false,
  debug: false,
}) {
  const merged = Object.assign(
    {},
    {indexPath: 'node_modules/quasar/src/css/index.sass'},
    opts
  )
  return sassMigrator(merged)
}

// yarn sass-migrator division --dry-run --migrate-deps node_modules/quasar/src/css/index.sass
