// rollup-plugin-sass-migrator/dist/sass-migrator.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.sassMigratorQuasar = exports.sassMigrator = void 0;
var child_process_1 = require("child_process");
var fs = __importStar(require("fs"));
function sassMigrator(opts) {
    if (opts === void 0) { opts = {
        indexPath: undefined,
        dryRun: false,
        debug: false,
    }; }
    if (!opts.indexPath) {
        console.error("The sass-migrator plugin requires a path to a SASS file.");
        return undefined;
    }
    if (!fs.existsSync('node_modules/sass-migrator/sass-migrator.js')) {
        console.error("The sass-migrator plugin requires sass-migrator to be installed.\nTry running yarn add -D sass-migrator");
        return undefined;
    }
    return {
        name: 'migrate-sass',
        buildStart: function () {
            if (opts.debug) {
                console.log("migrate-sass [".concat(opts.indexPath, "]"));
            }
            var args = [
                'node_modules/sass-migrator/sass-migrator.js',
                'division',
                '--migrate-deps',
                opts.indexPath,
            ];
            if (opts.dryRun) {
                args.splice(-1, 0, "--dry-run");
            }
            var ret = (0, child_process_1.spawnSync)("node", args);
            if (ret.status) {
                console.warn("There was an error migrating sass:\n".concat(ret.stderr.toString().trim()));
            }
            if (opts.debug) {
                console.log(ret.stdout.toString().trim());
            }
        }
    };
}
exports.sassMigrator = sassMigrator;
function sassMigratorQuasar(opts) {
    if (opts === void 0) { opts = {
        dryRun: false,
        debug: false,
    }; }
    var merged = Object.assign({}, { indexPath: 'node_modules/quasar/src/css/index.sass' }, opts);
    return sassMigrator(merged);
}
exports.sassMigratorQuasar = sassMigratorQuasar;
// yarn sass-migrator division --dry-run --migrate-deps node_modules/quasar/src/css/index.sass
